src/Components/App.js

```diff
@@ -1,34 +1,45 @@
 import React, { Component } from 'react'
 import '../CSS_styling/App.css';
-import movieData from '../movie-data';
 import Movies from './Movies';
 import NavBar from './NavBar';
 import Details from './Details'
+import { fetchMovies, getSingleMovie } from '../apiCalls';
 
 class App extends Component {
   constructor() {
    super()
    this.state = {
-     movieInfo: movieData.movies
+     movieInfo: [],
+     error: ''
    }
-   console.log("this.state.movieInfo here---", this.state.movieInfo)
+  }
+
+  componentDidMount = () => {
+    fetchMovies()
+    .then(data => this.setState({movieInfo: data.movies}))
+    .catch(error => this.handleError(error))
+  }
+
+  handleError = (error) => {
+      this.setState({error: `Oops! Something went front. Error reading: ${error}`})
   }
 
   displayMainPage = () => {
-    this.setState({movieInfo: movieData.movies})
+    this.componentDidMount()
   }
 
   findDetails = (event) => {
     let id = Number(event.target.id)
-    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
-    .then(response => response.json())
+    getSingleMovie(id)
     .then(data => this.setState({movieInfo: [data.movie]}))
+    .catch(error => console.log(error))
   }
 
   render() {
     return (
       <>
         <NavBar showMain={this.displayMainPage}/>
+        {this.state.error && <h1>{this.state.error}</h1>}
```

Comment:
It was cool figuring out when to use ternaries and when to use the logical AND operator